### PR TITLE
Improved closing tickets mechanism

### PR DESCRIPTION
**Pull request by konadrian**
Improved closing tickets mechanism

### CODE REVIEW

File: modules/HelpDesk/views/Detail.php [hunk 1]
```diff
@@ -32,7 +32,8 @@ public function loadJsConfig(App\Request $request)
 		parent::loadJsConfig($request);
 		foreach ([
 			'checkIfRecordHasTimeControl' => (bool) \App\Config::module('HelpDesk', 'CHECK_IF_RECORDS_HAS_TIME_CONTROL'),
-			'checkIfRelatedTicketsAreClosed' => (bool) \App\Config::module('HelpDesk', 'CHECK_IF_RELATED_TICKETS_ARE_CLOSED')
+			'checkIfRelatedTicketsAreClosed' => (bool) \App\Config::module('HelpDesk', 'CHECK_IF_RELATED_TICKETS_ARE_CLOSED'),
+			'closeTicketForStatus' => array_flip(\App\RecordStatus::getStates($request->getModule(), \App\RecordStatus::RECORD_STATE_CLOSED))
```
Comment (mariuszkrzaczkowski) [inline modules/HelpDesk/views/Detail.php]: ?

File: public_html/layouts/basic/modules/HelpDesk/resources/Edit.js [hunk 2]
```diff
@@ -119,7 +135,7 @@ Vtiger_Edit_Js(
 			}
 
 			quickCreateParams['data'] = relatedParams;
-			quickCreateParams['callbackFunction'] = function() {};
+			//		quickCreateParams['callbackFunction'] = function() {};
```
Comment (mariuszkrzaczkowski) [inline public_html/layouts/basic/modules/HelpDesk/resources/Edit.js]: ?

File: modules/HelpDesk/views/QuickCreateAjax.php [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+{
+	/**
+	 * {@inheritdoc}
+	 */
+	public function process(App\Request $request)
+	{
+		parent::process($request);
+		$moduleName = $request->getModule();
+		$viewer = $this->getViewer($request);
+		$viewer->assign('CHECK_IF_RECORDS_HAS_TIME_CONTROL', \App\Config::module($moduleName, 'CHECK_IF_RECORDS_HAS_TIME_CONTROL'));
```
Comment (mariuszkrzaczkowski) [inline modules/HelpDesk/views/QuickCreateAjax.php]: move to TPL

File: modules/HelpDesk/views/QuickCreateAjax.php [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+		$viewer->assign('CHECK_IF_RELATED_TICKETS_ARE_CLOSED', \App\Config::module($moduleName, 'CHECK_IF_RELATED_TICKETS_ARE_CLOSED'));
```
Comment (mariuszkrzaczkowski) [inline modules/HelpDesk/views/QuickCreateAjax.php]: move to TPL

File: modules/HelpDesk/views/QuickCreateAjax.php [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+		$viewer->assign('CLOSE_TICKET_FOR_STATUS', \App\Json::encode(array_flip(\App\RecordStatus::getStates($moduleName, \App\RecordStatus::RECORD_STATE_CLOSED))));
```
Comment (mariuszkrzaczkowski) [inline modules/HelpDesk/views/QuickCreateAjax.php]: move to TPL

File: layouts/basic/modules/HelpDesk/QuickCreate.tpl [hunk 6]
```diff
@@ -0,0 +1,7 @@
+{*<!-- {[The file is published on the basis of YetiForce Public License 3.0 that can be found in the following directory: licenses/LicenseEN.txt or yetiforce.com]} -->*}
+{strip}
+	{include file=\App\Layout::getTemplatePath('QuickCreate.tpl', 'Vtiger')}
+	<input value="{\App\Config::module($MODULE, 'CHECK_IF_RECORDS_HAS_TIME_CONTROL')}" type="hidden" id="checkIfRecordHasTimeControl">
```
Comment (rskrzypczak) [inline layouts/basic/modules/HelpDesk/QuickCreate.tpl]:
```suggestion
	<input value="{\App\Config::module($MODULE_NAME, 'CHECK_IF_RECORDS_HAS_TIME_CONTROL')}" type="hidden" id="checkIfRecordHasTimeControl">
```

**Pull request merged**